index.d.ts

```diff
@@ -0,0 +1,53 @@
+declare namespace camelcaseKeys {
+	interface Options {
+		/**
+		Recurse nested objects and objects in arrays.
+
+		@default false
+		*/
+		readonly deep?: boolean;
+
+		/**
+		Exclude keys from being camelCased.
+
+		@default []
+		*/
+		readonly exclude?: Array<string | RegExp>;
+	}
+}
+
+/**
+Convert object keys to camelCase using [`camelcase`](https://github.com/sindresorhus/camelcase).
+
+@param input - Object or array of objects to camelCase.
+
+@example
+```
+import camelcaseKeys = require('camelcase-keys');
+
+// Convert an object
+camelcaseKeys({'foo-bar': true});
+//=> {fooBar: true}
+
+// Convert an array of objects
+camelcaseKeys([{'foo-bar': true}, {'bar-foo': false}]);
+//=> [{fooBar: true}, {barFoo: false}]
+
+camelcaseKeys({'foo-bar': true, nested: {unicorn_rainbow: true}}, {deep: true});
+//=> {fooBar: true, nested: {unicornRainbow: true}}
+
+import minimist = require('minimist');
+
+const argv = minimist(process.argv.slice(2));
+//=> {_: [], 'foo-bar': true}
+
+camelcaseKeys(argv);
+//=> {_: [], fooBar: true}
+```
+*/
+declare function camelcaseKeys(
+	input: unknown,
+	options?: camelcaseKeys.Options
+): {[key: string]: unknown};
```

Comment:
Actually, might be better to have two overloads so we can preserve the input type.